ecl/hql/hqlfold.cpp

```diff
@@ -3584,8 +3584,40 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             break;
         }
     case no_random:
-        if (foldOptions & (HFOfoldimpure|HFOforcefold))
-            return createConstant(expr->queryType()->castFrom(true, (__int64)rtlRandom()));
+        {
+            if (foldOptions & (HFOfoldimpure|HFOforcefold))
+            {
+                if (expr->queryChild(0)->numChildren() == 0)
+                    return createConstant(expr->queryType()->castFrom(true, (__int64)rtlRandom()));
+
+                if (!expr->isConstant())
+                    break;
+
+                if (expr->queryChild(0)->queryName() == distributionUniformAtom)
```

Comment:
Better to assign queryChild(0) into a local variable.  Simplifies the following code, and makes it more efficient.

(Note all this code goes if implemented as a library function)